### PR TITLE
Frontend Overhaul blog post

### PR DESCRIPTION
**Pull request by xoscar**
Hello everyone,

Our team wrote a blog post about the frontend changes a couple of weeks ago, and there were some ideas on possibly having the post added to the open telemetry website as well. We have made some changes to make it vendor-free and better suited for the open telemetry website.

Let me know if you have any questions or concerns!

---

Preview: https://deploy-preview-1877--opentelemetry.netlify.app/blog/2022/frontend-overhaul/

### CODE REVIEW

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (melbgirl) [inline content/en/blog/2022/frontend-overhaul/index.md:74]: Rewording:
“Prior to re-architecting, the front end consisted of a Golang SSR app, which sent complete HTML to the browser for display. Every request and call redirected to the server so new information was shown.”

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (melbgirl) [inline content/en/blog/2022/frontend-overhaul/index.md:103]: Rewording:
“… that included the following:”

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (melbgirl) [inline content/en/blog/2022/frontend-overhaul/index.md:111]: “duringone” should be “during one”

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (melbgirl) [inline content/en/blog/2022/frontend-overhaul/index.md:115]: Keep “front end” or “frontend” consistent throughout. Same with “backend”

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (melbgirl) [inline content/en/blog/2022/frontend-overhaul/index.md:126]: Rewording:
“… was a way to instrument both sides of the Next.js app. To do this we had to connect the app twice to the same collector used by all the microservices.”

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (melbgirl) [inline content/en/blog/2022/frontend-overhaul/index.md:131]: Rewording:
“A simple backend solution was designed using the …”

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (melbgirl) [inline content/en/blog/2022/frontend-overhaul/index.md:141]: Rewording:
“As part of providing a better example for users, a manual instrumentation in the form of route middleware was added. This would catch the …”

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (melbgirl) [inline content/en/blog/2022/frontend-overhaul/index.md:146]: Rewording:
“The front end was a little trickier, as the rendering is server-side. We had to make sure to …”

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (melbgirl) [inline content/en/blog/2022/frontend-overhaul/index.md:151]: Rewording:
“…check the browser side, we then loaded the custom front end tracing module, which included creating the web tracer …”

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (melbgirl) [inline content/en/blog/2022/frontend-overhaul/index.md:154]: Rewording:
“The automatic front end instrumentation captures the most common user actions such as clicks, fetch requests, and page loads. In order to allow the browser side to interact with the collector, a config change is needed: enable incoming CORS requests from the web app. The resulting collector receivers looks something like this:”

**Review by melbgirl — CHANGES_REQUESTED** [review]
Mostly language changes/rewording to improve flow and readability.

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (svrnm) [inline content/en/blog/2022/frontend-overhaul/index.md:3]:
```suggestion
linkTitle: OTel Demo Frontend Overhaul
```

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (svrnm) [inline content/en/blog/2022/frontend-overhaul/index.md:6]: remove

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (svrnm) [inline content/en/blog/2022/frontend-overhaul/index.md:11]:
```suggestion
[OpenTelemetry Community Demo SIG](https://github.com/open-telemetry/opentelemetry-demo) which gives support to a set of instrumented
```

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (svrnm) [inline content/en/blog/2022/frontend-overhaul/index.md:39]:
```suggestion
A fun aspect of the work was the request to move the store from a "normal" store
```

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (svrnm) [inline content/en/blog/2022/frontend-overhaul/index.md:35]: what does SSR, CSR stand for? don't expect people to know that, introduce those abbreviation or link to a definition eg wikipedia

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (svrnm) [inline content/en/blog/2022/frontend-overhaul/index.md:161]: don't use a picture, where no picture is needed, some people can not see and some people might want to copy out the content of that picture easily.

**Review by svrnm — CHANGES_REQUESTED** [review]
thanks @xoscar!

@open-telemetry/demo-approvers please give this a look as well

**Comment by xoscar** [comment]
Hello @melbgirl & @svrnm thanks for your review! I really appreciate it 😄 

I have addressed all of your comments, let me know if there is anything else I can do. Cheers!

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (reyang) [inline content/en/blog/2022/frontend-overhaul/index.md:2]: "Frontend", "Front-end" or "Front end"?

https://en.wikipedia.org/wiki/Front-end_web_development

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (reyang) [inline content/en/blog/2022/frontend-overhaul/index.md:62]: Alphabetic ordering?

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (svrnm) [reply content/en/blog/2022/frontend-overhaul/index.md:2]: I think I saw "back-end" used a lot across our docs, so it would be "Front-end" then

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (reyang) [inline content/en/blog/2022/frontend-overhaul/index.md:28]:
```suggestion
The first thing we did was to get in contact with [Carter Socha](link to GitHub profile), the organizer of
```

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (reyang) [inline content/en/blog/2022/frontend-overhaul/index.md:65]: https://en.wikipedia.org/wiki/GRPC

```suggestion
a global gRPC definition. Persistent information is saved into a Postgres
```

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (reyang) [inline content/en/blog/2022/frontend-overhaul/index.md:68]:
```suggestion
including the front end, are connected to the same OpenTelemetry collector
```

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (reyang) [inline content/en/blog/2022/frontend-overhaul/index.md:69]: It sounds like traces and spans are different things, consider some rewording.

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (reyang) [inline content/en/blog/2022/frontend-overhaul/index.md:73]: Why uppercase "Server Side Rendered"?

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (reyang) [inline content/en/blog/2022/frontend-overhaul/index.md:117]:
```suggestion
As you can see in the diagram, the application has two major connectivity
```

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (reyang) [inline content/en/blog/2022/frontend-overhaul/index.md:146]:
```suggestion
the JavaScript code is executed.
```

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (reyang) [inline content/en/blog/2022/frontend-overhaul/index.md:156]:
```suggestion
Once the setup is complete, by loading the application from Docker and
```

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (reyang) [inline content/en/blog/2022/frontend-overhaul/index.md:172]: Consider link to GitHub profiles.

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (melbgirl) [inline content/en/blog/2022/frontend-overhaul/index.md:33]: If you want to use the acronym SSR you totally can. Best practice is to explain the acronym the first time you use it in a post, and then use the acronym after that. 
eg:
" ... away from Go Server-Side Renderer (SSR) to an architecture that included a browser-side client (Client-Side Renderer or CSR), as well as ..."

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (melbgirl) [reply content/en/blog/2022/frontend-overhaul/index.md:73]: Since it's been explained earlier in the post, the first time the acronym was used (I think it's line 33), you can use "SSR" here onwards.

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (melbgirl) [reply content/en/blog/2022/frontend-overhaul/index.md:2]: Sold!

**Comment by xoscar** [comment]
Hello @reyang thanks for your feedback, I think I got it all fixed now. Let me know if there is something else I can do! 😄

**Comment by reyang** [comment]
> Hello @reyang thanks for your feedback, I think I got it all fixed now. Let me know if there is something else I can do! 😄

Thanks @xoscar! Consider fixing the diagram, too.
In addition, the diagram has very small fonts (typefaces) which is hard to read, consider making it bigger.

image

**Review by reyang — APPROVED** [review]
LGTM with some minor suggestions.

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (melbgirl) [inline content/en/blog/2022/frontend-overhaul/index.md:37]: Should probably add the acronym for client-side render, ie:  "...client (client-side render or CSR), as ..."

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (melbgirl) [inline content/en/blog/2022/frontend-overhaul/index.md:91]: Would be good to provide a link to Olly Babiak's GitHub profile (or Twitter if GH not available).

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (melbgirl) [inline content/en/blog/2022/frontend-overhaul/index.md:112]: The SIG has been referred to as "demo SIG" throughout, suggest applying the lowercase 'd' here for consistency. Not a blocking item!

**Comment by xoscar** [comment]
Hey @svrnm & @melbgirl thanks for your comments, I think I have fixed all of them 😄

**Comment by svrnm** [comment]
TY @xoscar , minor NIT: the image `system-diagram.png` has some visual icons on the top right corner that make PHP invisible, can you fix that please?

**Comment by svrnm** [comment]
@open-telemetry/blog-approvers , please review & approve

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (cartermp) [inline content/en/blog/2022/frontend-overhaul/index.md:125]: One thing that would be really great to have here, and I won't block on this but I still think it'd be helpful, would be an elaboration on how instrumentation is plugging into next.js. Specifically, I see that there's a next.js middleware implementation that creates spans. That's cool! But why is it there, and what does it do? How does it get used? I think that would be _very_ interesting to people who are also getting into tracing with frontends.

**Review by cartermp — APPROVED** [review]

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (svrnm) [inline content/en/blog/2022/frontend-overhaul/index.md:74]: while github has a mermaid support, we do not have it in hugo, so you still need to put the picture here.

Comment (xoscar) [reply content/en/blog/2022/frontend-overhaul/index.md]: Ups, sorry about that, I just fixed it with my latest commit. Thanks for the heads up! 😄

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (cartermp) [reply content/en/blog/2022/frontend-overhaul/index.md:126]: Yep, that sounds neat! Maybe that could also make more sense as a readme or something in the demo repo itself. It's just something I got curious about since the post had me digging through code and it seemed cool but wasn't clear on the approach.

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (mic-max) [inline content/en/blog/2022/frontend-overhaul/index.md:19]: typo ```suggestion
prescribed in the official OTel documentation. More about the specific
```

**Review by mic-max — APPROVED** [review]

**Review by svrnm — APPROVED** [review]

**Comment by cartermp** [comment]
@open-telemetry/blog-approvers PTAL!

**Review by austinlparker — APPROVED** [review]

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (chalin) [inline content/en/blog/2022/frontend-overhaul/index.md:135]: Nit:
```suggestion
[Node.js SDK](https://www.npmjs.com/package/@opentelemetry/sdk-node).
```

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (chalin) [inline content/en/blog/2022/frontend-overhaul/index.md:65]: Would it make sense to make these links into the corresponding language SIG's landing page in the website?

**Review by chalin — COMMENTED** [review]
See inline for question and minor copy-edit.

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (xoscar) [reply content/en/blog/2022/frontend-overhaul/index.md:65]: Hello @chalin do you know where I can find the links to the SIG's landing page?

Comment (chalin) [reply content/en/blog/2022/frontend-overhaul/index.md:65]: I was referring to the listing of languages under https://opentelemetry.io/docs/instrumentation/

Comment (xoscar) [reply content/en/blog/2022/frontend-overhaul/index.md]: Done 😄

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (chalin) [inline content/en/blog/2022/frontend-overhaul/index.md:65]: Changes:

- Use local paths not external URLs here
- Alphabetical order of languge names: put .NET after C++ (as if it were named dotnet).

```suggestion
- [C++](/docs/instrumentation/cpp/)
- [.NET](/docs/instrumentation/net/)
- [Erlang/Elixir](/docs/instrumentation/erlang/)
- [Go](/docs/instrumentation/go/)
- [Node.js](/docs/instrumentation/js/)
- [PHP](/docs/instrumentation/php/)
- [Python](/docs/instrumentation/python/)
- [Ruby](/docs/instrumentation/ruby/)
- [Rust](/docs/instrumentation/rust/)
```

**Review by chalin — COMMENTED** [review]
See inline comment for a suggested change that will fix the build.

**Comment by xoscar** [comment]
@chalin I don't want to be pushy but, is there something else I can do to have this one merged? 😄

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (chalin) [inline content/en/blog/2022/frontend-overhaul/index.md:3]:
```suggestion
linkTitle: Front-end Overhaul Demo
```

**Review by chalin — APPROVED** [review]
One suggested (and optional) short-title tweak. Otherwise LGTM ✨

**Comment by chalin** [comment]
@xoscar - @svrnm and @cartermp will time the merge (and tweak the publish date?), once ready. I've given my final approval, thanks.

File: content/en/blog/2022/frontend-overhaul/index.md
Comment (chalin) [inline content/en/blog/2022/frontend-overhaul/index.md:4]:
```suggestion
date: 2022-11-07
draft: true
```

Comment (chalin) [reply content/en/blog/2022/frontend-overhaul/index.md:4]: Draft status as agreed during Comms meeting.

**Comment by cartermp** [comment]
Thanks @xoscar! We will un-draft this next Monday (discussed in Comms SIG) and have it get social media boostage.

**Comment by svrnm** [comment]
@xoscar see #2020

**Pull request merged**
